pkg/drivers/kic/oci/network.go

```diff
@@ -69,14 +69,49 @@ func digDNS(ociBin, containerName, dns string) (net.IP, error) {
 	return ip, nil
 }
 
+// gatewayIP inspects oci container to find a gateway IP string
+func gatewayIP(ociBin, containerName string) (string, error) {
+	rr, err := runCmd(exec.Command(ociBin, "container", "inspect", "--format", "{{.NetworkSettings.Gateway}}", containerName))
+	if err != nil {
+		return "", errors.Wrapf(err, "inspect gateway")
+	}
+	if gatewayIP := strings.TrimSpace(rr.Stdout.String()); gatewayIP != "" {
+		return gatewayIP, nil
+	}
+
+	// https://github.com/kubernetes/minikube/issues/11293
+	// need to check nested network
+	// check .NetworkSettings.Networks["cluster-name"].Gateway and then
+	// .NetworkSettings.Networks["bridge"|"podman"].Gateway
+	for _, network := range []string{containerName, defaultBridgeName(ociBin)} {
+		gatewayIP, err := networkGateway(ociBin, containerName, network)
+		if err != nil {
+			return "", err
+		}
+		if gatewayIP != "" {
+			return gatewayIP, nil
+		}
+	}
+	klog.Infof("Couldn't find gateway for container %s", containerName)
+	return "", nil
```

Comment:
shouldn't this be returning error ? instead of nil ?
how about we define an nil error before the `for loop`
```
        var gateErr error
	for _, network := range []string{containerName, defaultBridgeName(ociBin)} {....
	       if err != nil {
	      	       gateErr = err 
                }

	
	
	...}

return "", gateErr
```